extensions/arrow/src/main/java/io/deephaven/extensions/arrow/ArrowWrapperTools.java

```diff
@@ -157,11 +157,12 @@ public static QueryTable readFeather(@NotNull final String path) {
                 channel.position(block.getOffset());
 
                 final ByteBuffer metadataBuf = ByteBuffer.wrap(rawMetadataBuf, 0, block.getMetadataLength());
-                int numRead = channel.read(metadataBuf);
-                if (numRead != block.getMetadataLength()) {
-                    throw new IOException("Unexpected end of input trying to read block " + ii + " of '" + path + "'");
+                while (metadataBuf.hasRemaining()) {
+                    if (channel.read(metadataBuf) == -1) {
```

Comment:
We should probably throw if `<= 0` to ensure that each call is making progress.

Reply:
It's a valid point. Assuming the channel is in blocking mode, we _can_ throw an error on 0 (which _shouldn't_ occur)

> It is guaranteed, however, that if a channel is in blocking mode and there is at least one byte remaining in the buffer then this method will block until at least one byte is read.

I'll make the change.